integration-tests/transport/test_conservative_transport.py

```diff
@@ -0,0 +1,184 @@
+
+        u_expr = as_vector((u, w))
+
+        return u_expr
+
+    # Specify locations of the two Gaussians
+    xc1 = 5.*Lx/8.
+    zc1 = Hz/2.
+
+    xc2 = 3.*Lx/8.
+    zc2 = Hz/2.
+
+    def l2_dist(xc, zc):
+        return min_value(abs(x-xc), Lx-abs(x-xc))**2 + (z-zc)**2
+
+    lc = 2.*Lx/25.
+    m0 = 0.02
+
+    # Set the initial state from the configuration choice
+    if property == 'conservation':
+        f0 = 0.05
+
+        rho_t = 0.5
+        rho_b = 1.
+
+        rho_d_0 = rho_b + z*(rho_t-rho_b)/Hz
+
+        g1 = f0*exp(-l2_dist(xc1, zc1)/(lc**2))
+        g2 = f0*exp(-l2_dist(xc2, zc2)/(lc**2))
+
+        m_X_0 = m0 + g1 + g2
+    else:
+        f0 = 0.5
+        rho_b = 0.5
+
+        g1 = f0*exp(-l2_dist(xc1, zc1)/(lc**2))
+        g2 = f0*exp(-l2_dist(xc2, zc2)/(lc**2))
+
+        rho_d_0 = rho_b + g1 + g2
+
+        # Constant mass field
+        m_X_0 = m0 + 0*x
+
+    if space == 'diff_order_0':
+        VCG1 = FunctionSpace(mesh, 'CG', 1)
+        VDG1 = domain.spaces('DG1_equispaced')
+
+        suboptions = {'rho_d': RecoveryOptions(embedding_space=VDG1,
+                                               recovered_space=VCG1,
+                                               project_low_method='recover',
+                                               boundary_method=BoundaryMethod.taylor),
+                      'm_X': ConservativeRecoveryOptions(embedding_space=VDG1,
+                                                         recovered_space=VCG1,
+                                                         boundary_method=BoundaryMethod.taylor,
+                                                         rho_name='rho_d',
+                                                         orig_rho_space=V_rho)
+                      }
+    elif space == 'diff_order_1':
+        Vt_brok = FunctionSpace(mesh, BrokenElement(V_m_X.ufl_element()))
+        suboptions = {'rho_d': EmbeddedDGOptions(embedding_space=Vt_brok),
+                      'm_X': ConservativeEmbeddedDGOptions(rho_name='rho_d',
+                                                           orig_rho_space=V_rho)}
+    else:
+        suboptions = {}
+
+    opts = MixedFSOptions(suboptions=suboptions)
+
+    transport_scheme = SSPRK3(domain, options=opts, increment_form=False)
+    transport_methods = [DGUpwind(eqn, "m_X"), DGUpwind(eqn, "rho_d")]
+
+    # Timestepper
+    stepper = PrescribedTransport(eqn, transport_scheme, io, transport_methods, prescribed_transporting_velocity=u_t)
+
+    # Initial Conditions
+    stepper.fields("m_X").interpolate(m_X_0)
+    stepper.fields("rho_d").interpolate(rho_d_0)
+    u0 = stepper.fields("u")
+    u0.project(u_t(0))
+
+    m_X_init = Function(V_m_X)
+    rho_d_init = Function(V_rho)
+
+    m_X_init.assign(stepper.fields("m_X"))
+    rho_d_init.assign(stepper.fields("rho_d"))
+
+    return stepper, m_X_init, rho_d_init
+
+
+@pytest.mark.parametrize("space", ["same", "diff_order_0", "diff_order_1"])

```

Comment:
Could we call the `space` variable something like `pair_of_spaces`, and change `"same"` to `"same_order_1"`?